### ISSUE TITLE
clustering

### ISSUE THREAD

**Issue by incanus** [mapbox/ios-sdk-examples#66]
Per https://github.com/mapbox/mapbox-gl-native/issues/320#issuecomment-281695727, we should have a good example of symbol layer clustering. 

What I'd like to see: 

- Ability to toggle all clusters on & off
- Ability to tap on a cluster to have it zoom in to the extent of the clustered points (revealing more points)
- Ability to show a callout when tapping a cluster point
  - This might require some sort of interaction between a GL-backed symbol layer and an annotation view layer

Basically, let's: 
- Show what you can do _now_ with our clustering capabilities
- Get an idea of how important https://github.com/mapbox/mapbox-gl-native/issues/6181 is as a way to make this better.

**Comment by incanus** [comment]
Gathering some hidden lore from: 

- https://gist.github.com/boundsj/862b758cc9d1d022f1dac7e45a756cd7
- https://github.com/mapbox/mapbox-gl-native/issues/7387#issuecomment-266539546
- https://github.com/mapbox/mapbox-gl-native/issues/6609#issuecomment-251992910
- https://github.com/mapbox/mapbox-gl-style-spec/issues/519#issuecomment-252172929

Working on a pure-GL style spec clustering example with some interaction and animations. It also makes use of data-driven styling to do some interval-based styling on the cluster circle colors. 

![anigif-1487801145](https://cloud.githubusercontent.com/assets/17722/23234835/638a04ce-f921-11e6-9b0e-709b63ad4f35.gif)

**Comment by friedbunny** [comment]
Done in #66.

**Comment by Tamil-Arya** [comment]
Hi buddy, Is there any way to do assign different type of icon for each location while cluster ?

**Comment by friedbunny** [comment]
See https://github.com/mapbox/ios-sdk-examples/issues/79#issuecomment-290785380 for ways to get help with implementation questions.
